source_code/main_mcu/src/GUI/gui_prompts.c

```diff
@@ -1123,18 +1145,18 @@ RET_TYPE gui_prompts_get_user_pin(volatile uint16_t* pin_code, uint16_t stringID
     BOOL finished = FALSE;
     
     // Set current pin to 0000 or random
-    if (random_pin_feature_enabled == FALSE)
-    {
-        memset((void*)current_pin, 0, sizeof(current_pin));
-    }
-    else
+    if (show_pin_entry == TRUE && random_pin_feature_enabled == TRUE)
```

Comment:
only random when show pin is to TRUE? shouldn't it be the other way around?

Reply:
Do you mean changing the order of the condition like this ?
``` C
if (random_pin_feature_enabled == TRUE  && show_pin_entry == TRUE)
```
Or maybe the condition is not correct and I did not understand what your meant ? 😆